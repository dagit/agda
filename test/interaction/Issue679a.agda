module Issue679a where

data ty : Set where
  _`→_ : ty → ty → ty

⟦_⟧ : ty → Set
⟦ A `→ B ⟧ = ⟦ A ⟧ → ⟦ B ⟧

eq : ∀ (σ : ty) (a b : ⟦ σ ⟧) → Set
eq (A `→ B) f g = ∀ {a : ⟦ A ⟧} → eq B (f a) (g a)

eq-sym : ∀ σ {f g} (h : eq σ f g) → eq σ g f
eq-sym (A `→ B) h with B
... | B' = {!!}
