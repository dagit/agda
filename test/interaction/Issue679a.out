Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue679a (Issue679a.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished Issue679a.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : {a : ⟦ A ⟧} → eq B' (.g a) (.f a)\n" nil)
((last . 1) . (agda2-goals-action '(0)))
Agda2> (agda2-give-action 0 "λ {a} → ?")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?1 : eq B' (.g a) (.f a)\n" nil)
((last . 1) . (agda2-goals-action '(1)))
Agda2> 